src/app.rs

```diff
@@ -174,6 +174,9 @@ impl<'a, 'b> Application<'a, 'b> {
             };
 
             for event in events {
+                if let &Event::WindowEvent { event: WindowEvent::Closed, .. } = &event {
```

Comment:
So how could the state handle a window close event now?

Reply:
I could rearrange this so that the state receives this.  It just seems odd for our game window to by default *not* respond to a close event.  It confused me for a second when it happened so I'm sure it'd confuse someone not familiar with the engine.

Also it's worth noting that if you want this to be handled properly at all times you have to duplicate the close code for every state your game goes through.  Most of our examples have just one state so this isn't as apparent.

Reply:
I see it makes sense, but at least giving the game a notification would be great. Maybe also just call a method which is always when states are stopped? Not sure if we have that already.